### PR TITLE
secrets build from source

### LINKED ISSUE #3247: Extend build from source to SECRETS

### PR DESCRIPTION
**Pull request by gearama**
closes #3247 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt [hunk 1]
```diff
@@ -112,3 +129,5 @@ az_rtti_setup(
   SECURITY_KEYVAULT_SECRETS
   "azure/keyvault/secrets/rtti.hpp"
 )
+
+unset(BUILD_SOURCE_DEPS CACHE)
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:133]: what's this for?

new line missing

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt [hunk 2]
```diff
@@ -9,10 +9,19 @@ az_vcpkg_integrate()
 cmake_minimum_required (VERSION 3.13)
 project(azure-security-keyvault-secrets LANGUAGES CXX)
 
+# Compile Options
+option(BUILD_SOURCE_DEPS "build source dependencies" OFF)
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:13]: Update this PR on top of latest main

**Comment by gearama** [comment]
the core versions will be decided in the next PR

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt (hunk 2 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:13]: Didn't it change the name of this Option?

**Review by vhvb1989 — APPROVED** [review]
Check again the name of the options.  Do we still have the old name in some places?

**Pull request merged**